### PR TITLE
fix(charts): Hide Values greater than Max Y Axis Bound on Mixed Time Series with Bar series

### PR DESCRIPTION
**Pull request by Antonio-RiveroMartnez**
### SUMMARY
- When Bar chart is used as serie type, we need to hide values that are greater than the max Y Axis Bound so users don't get inconsistency problems with other serie types. User can still see the value using the tooltip.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:
error

After:
test
tooltip_1
tooltip_2

### TESTING INSTRUCTIONS
1. Create a mixed time series chart
2. Make sure value labels are on
3. In your second query, set the Y-Axis Bound Maximum to lower than the initial value in the chart

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/plugins/plugin-chart-echarts/src/utils/series.ts [hunk 1]
```diff
@@ -313,3 +314,19 @@ export function getAxisType(dataType?: GenericDataType): AxisType {
...
+    max?: number;
+  } = {},
+) {
+  const { max } = config;
+  // Only apply this logic if there's a MAX set in the controls
+  const shouldHideIfOverMax = !!max || max === 0;
+
+  return new NumberFormatter({
+    formatFunc: value =>
+      `${shouldHideIfOverMax ? (value > max ? '' : value) : value}`,
```
Comment (diegomedina248) [inline superset-frontend/plugins/plugin-chart-echarts/src/utils/series.ts:329]: `shouldHideIfOverMax && value > max ? '' : value`?

**Review by diegomedina248 — APPROVED** [review]

File: superset-frontend/plugins/plugin-chart-echarts/src/utils/series.ts [hunk 2]
```diff
@@ -313,3 +314,18 @@ export function getAxisType(dataType?: GenericDataType): AxisType {
   }
   return 'category';
 }
+
+export function getOverMaxHiddenFormatter(
```
Comment (stephenLYZ) [inline superset-frontend/plugins/plugin-chart-echarts/src/utils/series.ts:318]: Can we add some test for this util?

Comment (Antonio-RiveroMartnez) [reply superset-frontend/plugins/plugin-chart-echarts/src/utils/series.ts:318]: Done 👍

**Review by rusackas — APPROVED** [review]
LGTM

**Pull request merged**